packages/b/backward-cpp/xmake.lua

```diff
@@ -1,43 +1,95 @@
+
     if is_plat("mingw") then
-        add_patches("v1.6", path.join(os.scriptdir(), "patches", "v1.6", "link_to_imagehlp.patch"), "0a135b6d68970ff6609a3eb4deb2b10c317eee15ba980eb178b93402a97c957c")
+        add_patches("v1.6", "patches/v1.6/link_to_imagehlp.patch", "0a135b6d68970ff6609a3eb4deb2b10c317eee15ba980eb178b93402a97c957c")
 
         if is_arch("i386") then
-            add_patches("v1.6", path.join(os.scriptdir(), "patches", "v1.6", "fix_32bit_ssize_t_typedef.patch"), "fb372fe5934984aecb00b3153f737f63a542ff9359d159a9bcb79c5d54963b42")
+            add_patches("v1.6", "patches/v1.6/fix_32bit_ssize_t_typedef.patch", "fb372fe5934984aecb00b3153f737f63a542ff9359d159a9bcb79c5d54963b42")
         end
     end
 
+    if is_plat("windows", "mingw") then
+        add_syslinks("psapi", "dbghelp")
+    elseif is_plat("linux", "bsd", "android") then
+        add_syslinks("dl", "m")
+    end
+
     add_deps("cmake")
 
-    on_load("linux", "mingw@msys", "macos", function (package)
-        if package:config("bfd") then
-            package:add("deps", "binutils")
+    on_load(function (package)
+        local stack_walking = package:config("stack_walking")
+        if stack_walking == "libunwind" then
+            package:add("deps", "libunwind")
         end
-    end)
 
-    on_install("linux", "mingw", "macosx", "windows|!arm*", function (package)
-        if package:config("bfd") and package:is_plat("linux", "mingw@msys", "macos") then
+        local stack_details = package:config("stack_details")
+        if stack_details == "dwarf" then
+            package:add("deps", "libdwarf", "libelf")
+        elseif stack_details == "dw" then
+            package:add("deps", "elfutils")
+        elseif stack_details == "bfd" then
+            package:add("deps", "binutils")
             package:add("syslinks", "bfd")
-            package:add("defines", "BACKWARD_HAS_BFD=1")
         end
-        local configs = {"-DBACKWARD_TESTS=OFF"}
+
+        package:add("defines", "BACKWARD_HAS_UNWIND=" .. (stack_walking == "unwind" and "1" or "0"))
+        package:add("defines", "BACKWARD_HAS_LIBUNWIND=" .. (stack_walking == "libunwind" and "1" or "0"))
+        package:add("defines", "BACKWARD_HAS_BACKTRACE=" .. (stack_walking == "backtrace" and "1" or "0"))
+
+        package:add("defines", "BACKWARD_HAS_BACKTRACE_SYMBOL=" .. (stack_details == "backtrace_symbol" and "1" or "0"))
+        package:add("defines", "BACKWARD_HAS_DW=" .. (stack_details == "dw" and "1" or "0"))
+        package:add("defines", "BACKWARD_HAS_BFD=" .. (stack_details == "bfd" and "1" or "0"))
+        package:add("defines", "BACKWARD_HAS_DWARF=" .. (stack_details == "dwarf" and "1" or "0"))
+        package:add("defines", "BACKWARD_HAS_PDB_SYMBOL=" .. (is_plat("windows") and "1" or "0"))
+    end)
+
+    on_install("windows|!arm64", function (package)
```

Comment:
this should be "!windows or windows|!arm64"

Reply:
you can open pr to improve it